libcxx/include/__format/format_arg.h

```diff
@@ -250,29 +250,79 @@ class __basic_format_arg_value {
     __handle __handle_;
   };
 
+  // Visiting a value for formatting is dispatched via a function pointer to prevent
+  // unused formatters from being instantiated. This can lead to significant code size 
+  // reductions on statically linked targets esp. at lower optimization levels.
+
+  void (__basic_format_arg_value::*__format_)(
+      basic_format_parse_context<_CharT>& __parse_ctx, _Context& __ctx, bool __parse) const;
```

Comment:
This adds a new member to the class and changes its size. This affects the size of `basic_format_arg` and is an ABI break.

I have not started the GH CI, the Buildkite CI fails to use `std::format` with the current changes.

Reply:
Then there is no way to fix your code gen without an ABI break. 

A similar change could be made in `std::make_format_args`, where it generates the minimal visitor function while it still has all the type information. That would still change the size of some types although more palatable to maintain ABI compatibility. __vformat and the arg store would have two versions.

Reply:
libc++ promises ABI stability to our users and breaking the ABI needs a very good reason. When there is not a very good reason we make these changes in the unstable ABI. So there changes need some `ifdef`s. Before doing that I like to see some numbers so we can evaluate the merits of this change.

Reply:
What form would you like the numbers in? e.g., original code is this size/performance vs various tweaks with this size/performance.

I will isolate each change as its own build flag so we can properly analyze its overhead. I know you have existing benchmarks but I will use a simple timing loop instead to better isolate changes.

Reply:
yes size/performance before and with the changes. If you use your own files can you add them to the review? We can remove them later but it makes it easier to test the changes locally.